src/lib/pdfconverter.cc

```diff
@@ -963,9 +963,6 @@ void PdfConverterPrivate::endPrintObject(PageObject & obj) {
 	pageFormElements.clear();
 
 	if (obj.web_printer != 0) {
-		delete obj.web_printer;
```

Comment:
I'm not sure about this, what happens in the call from line 887? I don't want to cause one more breakage due to a code path not being considered.

Reply:
As far as I understand the code (I have looked at it for just a few hours so I might be wrong) the loop at printDocument() deletes the web_printer after each object is used. The call from 887 (at beginPrintObject) deletes the web_printer of previous object when the loop advances, and finally the call from line 1012 clears the web_printer of the last object.

The problem is that when the "copies" loop (at 995) advances to print the next page, web_printers were already deleted so the next copy has nothing to print, and thus --copies always produces a single copy.

My solutions was to defer the web_printer deletion until all pages were processed, but of course it might be not the best solution (I might even be completely wrong!).

Be free to reject the PR if you have doubts about its correctness, as I said I have only looked at the code for a few hours. In any case, we have this solution working on a system that produces more than 50.000 PDFs per day (could not wait for official release because of the importance of the --copies feature to us), and it is working ok so far.

Reply:
That's good to know. I'll review it in more detail this weekend and merge it :+1:

Reply:
Sorry about the delay, I've been a bit busy but I'll look at it this week.